packages/calendar/src/Calendar.ts

```diff
@@ -0,0 +1,449 @@
+            weekday: 'long',
+            year: 'numeric',
+            month: 'long',
+            day: 'numeric',
+        });
+
+        // TODO: Translate "Today" and "selected"
+        const todayTitle = isToday ? 'Today, ' : '';
+        const selectedTitle = isToday ? ' selected' : '';
+
+        const title: string = this._capitalFirstLetter(
+            `${todayTitle}${currentDayTitle}${selectedTitle}`
+        );
+
+        return html`
+            <td
+                role="gridcell"
+                class="spectrum-Calendar-tableCell"
+                title=${title}
+                tabindex=${ifDefined(!isOutsideMonth ? '-1' : undefined)}
+                aria-disabled=${isOutsideMonth || this.disabled}
+                aria-selected=${isSelected}
+            >
+                <span
+                    role="presentation"
+                    class="spectrum-Calendar-date ${classMap(dayClasses)}"
+                    @click=${() => this.handleDayClick(calendarDate)}
+                >
+                    ${this._formatNumber(calendarDate.day)}
+                </span>
+            </td>
+        `;
+    }
+
+    public handlePreviousMonth(): void {
+        this._currentDate = startOfMonth(this._currentDate).subtract({
+            months: 1,
+        });
+    }
+
+    public handleNextMonth(): void {
+        this._currentDate = startOfMonth(this._currentDate).add({ months: 1 });
+    }
+
+    public handleDayClick(calendarDate: CalendarDate): void {
+        this.selectedDate = calendarDate.toDate(this._timeZone);
+        this._setCurrentCalendarDate();
+
+        this.dispatchEvent(
+            new CustomEvent('change', {
+                bubbles: true,
+                cancelable: true,
+                composed: true,
+                detail: this.selectedDate,
+            })
+        );
+    }
+
+    private _getWeeksInCurrentMonth(): number {
+        return getWeeksInMonth(this._currentDate, this._locale);
+    }
+
+    /**
+     * Returns data for the days of the week, starting with the first day of the week according to the defined locale
+     * (Sunday, Monday, etc.)
+     */
+    private _getWeekdays(): CalendarWeekday[] {
+        return [...new Array(daysInWeek).keys()].map((dayIndex) => {
+            const weekStart = startOfWeek(this._currentDate, this._locale);
+            const date = weekStart.add({ days: dayIndex });
+
+            return {
+                narrow: this._formatDate(date, { weekday: 'narrow' }),
+                long: this._capitalFirstLetter(
+                    this._formatDate(date, { weekday: 'long' })
+                ),
+            };
+        });
+    }
+
+    private _setLocale(): void {
+        this._locale = this._languageResolver.language;
+    }
+
+    private _setInitialCalendarDate(): void {
+        this._currentDate = this.today;
+    }
+
+    private _setCurrentCalendarDate(): void {
+        if (this.selectedDate) {
```

Comment:
Please you use an early return pattern in these methods.

Reply:
But here I don't return anything, I only do something if there really is a `selectedDate`. How do you think I could improve this part?

Reply:
Something like with an early return pattern would be.
```
  if (!this.selectedDate) {
    return; // Early return if selectedDate is falsy (null, undefined, etc.)
  }

  this.selectedDate = new Date(this.selectedDate);

  if (!this._isValidDate(this.selectedDate)) {
    this.selectedDate = undefined;
    return; // Early return if the selectedDate is not a valid date
  }

  this._currentDate = this._toCalendarDate(this.selectedDate);
```